test/mapmatch.cc

```diff
@@ -0,0 +1,234 @@
+      "isochrone": {"max_contours": 4,"max_distance": 25000.0,"max_locations": 1,"max_time": 120},
+      "max_avoid_locations": 50,"max_radius": 200,"max_reachability": 100,
+      "multimodal": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 0.0,"max_matrix_locations": 0},
+      "pedestrian": {"max_distance": 250000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50,"max_transit_walking_distance": 10000,"min_transit_walking_distance": 1},
+      "skadi": {"max_shape": 750000,"min_resample": 10.0},
+      "trace": {"max_distance": 200000.0,"max_gps_accuracy": 100.0,"max_search_radius": 100,"max_shape": 16000},
+      "transit": {"max_distance": 500000.0,"max_locations": 50,"max_matrix_distance": 200000.0,"max_matrix_locations": 50},
+      "truck": {"max_distance": 5000000.0,"max_locations": 20,"max_matrix_distance": 400000.0,"max_matrix_locations": 50}
+    }
+  })");
+
+  std::string json_escape(const std::string& unescaped) {
+    std::stringstream ss;
+    baldr::json::OstreamVisitor v(ss);
+    v(unescaped);
+    std::string escaped = ss.str().substr(1);
+    escaped.pop_back();
+    return escaped;
+  }
+
+  template <class container_t>
+  std::string print(const container_t& container) {
+    std::string output;
+    for(const auto& e : container)
+      output += std::to_string(e) + ",";
+    return output;
+  }
+
+  template <typename T>
+  struct ring_queue_t {
+    ring_queue_t(size_t limit):limit(limit), i(0) {
+      v.reserve(limit);
+    }
+    void emplace_back(T&& t){
+      if(v.size() < limit) v.emplace_back(t);
+      else v[i] = t;
+      i = (i + 1) % limit;
+    };
+    const T& front() const { return i < v.size() ? v[i] : v[0]; }
+    const T& back() const { return v[i - 1]; }
+    size_t size() const { return v.size(); }
+    bool full() const { return v.size() == limit; }
+
+    size_t limit, i;
+    std::vector<T> v;
+
+    using iterator = typename std::vector<T>::iterator;
+    using const_iterator = typename std::vector<T>::const_iterator;
+    iterator begin() { return v.begin(); }
+    const_iterator begin() const { return v.begin(); }
+    iterator end() { return v.end(); }
+    const_iterator end() const { return v.end(); }
+  };
+
+  std::vector<midgard::PointLL> resample_at_1hz(const boost::property_tree::ptree& edges, const std::vector<midgard::PointLL>& shape) {
+    std::vector<midgard::PointLL> resampled;
+    float time_remainder = 0.0;
+    for(const auto& edge_item: edges) {
+      const auto& edge = edge_item.second;
+      //get the portion of the shape that applies to this edge
+      std::vector<midgard::PointLL> edge_shape(shape.cbegin() + edge.get<size_t>("begin_shape_index"),
+        shape.cbegin() + edge.get<size_t>("end_shape_index") + 1);
+      //get the speed of this edge
+      auto meters = midgard::Polyline2<PointLL>::Length(edge_shape);
+      auto speed = (edge.get<float>("speed") * 1e3) / 3600.f;
```

Comment:
yep it assumes kph, is that a wrong assumption? i didnt set it in `directions_options` assuming kph would come back, i figured that was koshure. would you rather i set it on `directions_options`?